### PR TITLE
Docs: Updates Salesforce Source Connector Docs (Removes inapp page, adds <HideInUI>, adds Limitations & Troubleshooting)

### PR DESCRIPTION
**Pull request by pnilan**
## What
Updates for Salesforce source connector docs pages. This update includes addition of new collapsible troubleshooting section, as well as the replacement of the "inapp" docs with the new <HideInUI> tags per Issue #32069.

## How
- Removed `in-app` page
- Added `HideInUI` tags to all sections other than `Prerequisites` and `Setup Guide`.
- `env` tags handle sections to be displayed only in OSS vs Cloud pages.
- Added Limitations & Troubleshooting section, per new [suggested approach](https://docs.google.com/document/d/1blZ5G6kd8CA2Ym83F7GqPLz7esqT7iX2bW2aG4jW9pU/edit).

### Main page vs In-app OSS (Example)
Shopify docs difference

### Limitations & Troubleshooting Section (Example)
Limitations screenshot

### CODE REVIEW

File: docs/integrations/sources/salesforce.md [hunk 1]
```diff
@@ -132,20 +166,28 @@ Salesforce allows extracting data using either the [BULK API](https://developer.
 
 More information on the differences between various Salesforce APIs can be found [here](https://help.salesforce.com/s/articleView?id=sf.integrate_what_is_api.htm&type=5).
 
-:::info Force Using Bulk API
+:::note Force Using Bulk API
```
Comment (katmarkham) [inline docs/integrations/sources/salesforce.md:169]: this is rendering weirdly

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/salesforce.md (hunk 1 above)
Comment (pnilan) [reply docs/integrations/sources/salesforce.md:169]: > this is rendering weirdly

@katmarkham Is this happening via Docusaurus or in the markdown preview in GitHub? These are [Docusaurus admonitions](https://docusaurus.io/docs/markdown-features/admonitions) so I don't think they'll render in GitHub markdown.

This is what I see when running Docusaurus locally:

Screenshot 2023-11-10 at 10 40 48 AM

File: docs/integrations/sources/salesforce.md [hunk 2]
```diff
@@ -103,12 +119,30 @@ Airbyte allows exporting all available Salesforce objects dynamically based on:
...
 
 Salesforce allows extracting data using either the [BULK API](https://developer.salesforce.com/docs/atlas.en-us.236.0.api_asynch.meta/api_asynch/asynch_api_intro.htm) or [REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/intro_what_is_rest_api.htm). To achieve fast performance, Salesforce recommends using the BULK API for extracting larger amounts of data (more than 2,000 records). For this reason, the Salesforce connector uses the BULK API by default to extract any Salesforce objects, unless any of the following conditions are met:
 
 - The Salesforce object has columns which are unsupported by the BULK API, like columns with a `base64` or `complexvalue` type
-- The Salesforce object is not supported by BULK API. In this case we sync the objects via the REST API which will occasionalyl cost more of your API quota. This list of objects was obtained experimentally, and includes the following objects: 
+- The Salesforce object is not supported by BULK API. In this case we sync the objects via the REST API which will occasionalyl cost more of your API quota. This list of objects was obtained experimentally, and includes the following objects:
```
Comment (katmarkham) [inline docs/integrations/sources/salesforce.md:145]: *occasionally

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/salesforce.md (hunk 2 above)
Comment (pnilan) [reply docs/integrations/sources/salesforce.md]: Thanks for the catch. Updated.

**Review by katmarkham — APPROVED** [review]

**Pull request merged**